### PR TITLE
Add support for elrond-go feat/rosetta

### PR DESCRIPTION
**Pull request by andreibancioiu**
For API, new fields on _transaction_:
 - `ProcessingTypeOnSource`
 - `ProcessingTypeOnDestination`
- `InitiallyPaidFee`
- `IsRefund`

For API, new fields on _miniblock_:
 - `ProcessingType`
 - `ConstructionType`

Improved support for block & account querying.

Extended `chain handler` interface.

### CODE REVIEW

**Review by bogdan-rosianu — APPROVED** [review]

**Review by schimih — APPROVED** [review]

File: data/transaction/apiTransactionResult.go [hunk 1]
```diff
@@ -11,6 +11,9 @@ import (
 type ApiTransactionResult struct {
 	Tx                                data.TransactionHandler   `json:"-"`
 	Type                              string                    `json:"type"`
+	ProcessingTypeOnSource            string                    `json:"processingTypeOnSource,omitempty"`
+	ProcessingTypeOnDestination       string                    `json:"processingTypeOnDestination,omitempty"`
+	HashBytes                         []byte                    `json:"-"`
```
Comment (miiu96) [inline data/transaction/apiTransactionResult.go:16]: maybe you can move `HashBytes` field after `Hash`  field in order to keep field with the same type togherder

File: data/transaction/apiTransactionResult.go [hunk 2]
```diff
@@ -52,6 +55,8 @@ type ApiTransactionResult struct {
 	Operation                         string                    `json:"operation,omitempty"`
 	Function                          string                    `json:"function,omitempty"`
 	IsRelayed                         bool                      `json:"isRelayed,omitempty"`
+	IsRefund                          bool                      `json:"isRefund,omitempty"`
+	InitiallyPaidFee                  string                    `json:"initiallyPaidFee,omitempty"`
```
Comment (miiu96) [inline data/transaction/apiTransactionResult.go:59]: also here `InitiallyPaidFee` can be more near `Function` field

**Review by schimih — DISMISSED** [review]

File: data/transaction/apiTransactionResult.go (hunk 1 above)
Comment (andreibancioiu) [reply data/transaction/apiTransactionResult.go]: Fixed.

File: data/transaction/apiTransactionResult.go (hunk 2 above)
Comment (andreibancioiu) [reply data/transaction/apiTransactionResult.go]: Fixed.

**Review by miiu96 — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**